net/netdev/netdev_carrier.c

```diff
@@ -62,8 +62,12 @@ int netdev_carrier_on(FAR struct net_driver_s *dev)
 {
   if (dev)
     {
-      dev->d_flags |= IFF_RUNNING;
-      netlink_device_notify(dev);
+      if (!IFF_IS_RUNNING(dev->d_flags))
```

Comment:
So we expect `netdev_carrier_on` to be called multiple times in a row? I mean that I do not see any harm in this code, but do not see explicit benefits.

Reply:
Yes, this issue can also be solved on the driver side, but I think if which can be fix inside netdev_carrier_*, it will also bring benefits to other drivers, because it is unnecessary to report more of the same events to netlink

Reply:
Yes and no :) I mean if 2 out of 100 drivers can call `netdev_carrier_on` in a row, then we add check (extra code) for 98 cases that do not need it. Now if driver call it once then event is reported once to netlink, so no issue.
Anyway this will lead to a code duplication in 2 (or more) cases.
I do not have anything against this change, so we can keep it.